ui/src/app/views/registration-dashboard.tsx

```diff
@@ -213,24 +227,71 @@ export class RegistrationDashboard extends React.Component<RegistrationDashboard
     }
   }
 
+  async setAllModulesBypassed(isBypassed: boolean) {
+    this.setState({bypassInProgress: true});
+
+    // TypeScript enum iteration is nonfunctional
+    // so just copy the whole list
+    const modules = [
+      AccessModule.COMPLIANCETRAINING,
+      AccessModule.ERACOMMONS,
+      AccessModule.TWOFACTORAUTH,
+      AccessModule.DATAUSEAGREEMENT,
+      AccessModule.BETAACCESS
+    ];
+
+    for (const module of modules) {
+      await profileApi().unsafeSelfBypassAccessRequirement({
+        moduleName: module,
+        isBypassed: isBypassed
+      });
+    }
+
+    // Call the callback (which refreshes the user's profile).
+    this.setState({bypassInProgress: false, bypassActionComplete: true});
+  }
+
   render() {
-    const {taskCompletionMap, trainingWarningOpen} = this.state;
+    const {bypassActionComplete, bypassInProgress, trainingWarningOpen} = this.state;
     const {betaAccessGranted, eraCommonsError, trainingCompleted} = this.props;
+    // todo: move this to the state
+    const canUnsafeSelfBypass = serverConfigStore.getValue().unsafeAllowSelfBypass;
+
+    const anyBypassActionsRemaining = !(this.allTasksCompleted() && betaAccessGranted);
+
     return <div style={styles.registrationPage}
                 data-test-id='registration-dashboard'>
+      {bypassInProgress && <SpinnerOverlay />}
       <div style={styles.mainHeader}>Researcher Workbench</div>
       <div style={{...styles.mainHeader, fontSize: '18px', marginBottom: '1rem'}}>
         <ClrIcon shape='warning-standard' class='is-solid'
                  style={{color: colors.white, marginRight: '0.3rem'}}/>
         In order to get access to data and tools please complete the following steps:
       </div>
-      {!betaAccessGranted && <div data-test-id='beta-access-warning'
-                                  style={{...baseStyles.card, ...styles.warningModal}}>
-        <ClrIcon shape='warning-standard' class='is-solid'
-                 style={styles.warningIcon}/>
-        You have not been granted beta access. Please contact support@researchallofus.org.
-      </div>}
-
+      {canUnsafeSelfBypass &&
+        <div data-test-id='self-bypass'
+             style={{...baseStyles.card, ...styles.warningModal}}>
+          {bypassActionComplete &&
+            <span>Bypass action is complete. Reload the page to continue.</span>}
+          {!bypassActionComplete && <span>
+            [Test environment] Self-service bypass is enabled:&nbsp;
+            {anyBypassActionsRemaining &&
+              <Button onClick={() => this.setAllModulesBypassed(true)}
+                      disabled={bypassInProgress}>Bypass all</Button>}
+            {!anyBypassActionsRemaining &&
+              <Button onClick={() => this.setAllModulesBypassed(false)}
+                      disabled={bypassInProgress}>Un-bypass all</Button>}
+          </span>
+          }
+        </div>
+      }
+      {!betaAccessGranted && !canUnsafeSelfBypass &&
```

Comment:
would we still wanna show this even if the user *can* selfbypass?  i am of the opinion that we should not hide user flow pieces for users with adminish privileges, in case someone wants to develop this piece or see it in action